### DISCUSSION TITLE
Supabase import functions are cached, causing issues

### DISCUSSION THREAD

**Discussion by adamcohenhillel** [supabase/supabase#17322] [Questions]
This is a follow-up discussion to https://github.com/orgs/supabase/discussions/17311, as it was closed.

All of a sudden, invoking functions in a specific project stopped working (it worked perfectly before). I tried reverting changes to different known versions that used to work, I tried deleting all functions and deploying them again, I tried to deploy a simple function just for testing - and it keeps failing. (The frontend does work with a different page).

These are the symptoms: 
The error on the client side is:
`Access to fetch at 'https://XXX.supabase.co/functions/v1/YYY' from origin 'http://localhost:3000' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.`

And on the edge function side:
![image](https://github.com/supabase/supabase/assets/633840/bdbda07b-b248-47b0-8084-6083b51680d8)

and:
![image](https://github.com/supabase/supabase/assets/633840/29288a7e-fc11-4e7c-8125-d34320b84a2f)

From discussing with the Supabase team, it seems like there is an issue with the package "@azure/core-rest-pipeline" - but this package is not a direct dependency of the code (deploying a simple function from the Supabase tutorials also fails with the same error). The package versions haven't changed either.

I reckon there is some sort of cache on the supabase end that I can't seem to clear - any ideas?

Supabase team - I am opening this discussion because we are in a critical need to solve this, and I can't wait for you to answer me privately, so **please** - leave this open.

**Comment by silentworks** [comment]
Is this only happening to that particular project or any project you create and test the function in?

**Comment by adamcohenhillel** [comment]
Resolution: move away from supabase functions.
